src/rgw/rgw_compression.cc

```diff
@@ -121,15 +121,19 @@ int RGWGetObj_Decompress::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len
   }
 
   cur_ofs += bl_len;
-
-  off_t ch_len = std::min<off_t>(out_bl.length() - q_ofs, q_len);
-  r = next->handle_data(out_bl, q_ofs, ch_len);
-  if (r < 0) {
-    lderr(cct) << "handle_data failed with exit code " << r << dendl;
-    return r;
+  //must check, for out_bl maybe empty when codes reach here, and q_len maybe 0 also
+  if (out_bl.length() && q_len) {
+    //first slice or last slice with bytes < cct->_conf->rgw_max_chunk_size
+    off_t ch_len = std::min<off_t>(out_bl.length() - q_ofs, q_len);
+    r = next->handle_data(out_bl, q_ofs, ch_len);
```

Comment:
@fangyuxiangGL As I understand, the problem is when we try to next->handle_data on 0 sized data. Maybe check ch_len>0 will be more natural?

Reply:
@aclamk 
yeah, root cause is that no bytes need to be handled, so ch_len > 0 may be more comprehensive and concise. But I think out_bl.length() && q_len